spacy_llm/models/langchain/model.py

```diff
@@ -0,0 +1,130 @@
+import warnings
+from typing import Any, Callable, Dict, Iterable, Optional, Type, TypeVar
+
+from spacy.util import SimpleFrozenDict
+
+from ...compat import has_langchain, langchain
+from ...registry import registry
+
+# Type of prompts returned from Task.generate_prompts().
+_PromptType = TypeVar("_PromptType")
+# Type of responses returned from query function.
+_ResponseType = TypeVar("_ResponseType")
+
+
+class LangChain:
+    def __init__(
+        self,
+        name: str,
+        api: str,
+        config: Dict[Any, Any],
+        query: Callable[
+            ["langchain.llms.base.BaseLLM", Iterable[_PromptType]],
+            Iterable[_ResponseType],
```

Comment:
This usage of `TypeVar`s doesn't really make sense. It's pretty much meaningless to do this unless the `TypeVar` is constrained and the query is generic. 

If `LangChain` as a class was generic and had some way to fill in these prompt/response types with e.g. a Pydantic model that was going to validate them then it would make more sense. 

But as this works now, they might as well be `Any`.

I think the same can be said for the `hf` models

Reply:
Fair point. I believe the `TypeVar`s were still in there from an earlier, different design. Removed in https://github.com/explosion/spacy-llm/pull/176/commits/57bc905f5cd505ce36f4fa3d89ab8f51092147e3.